src/frontend/hooks/useSyncState.ts

```diff
@@ -122,19 +141,33 @@ class SyncStore {
   }
 
   #onSyncState = (state: SyncState) => {
-    // Indicates whether data syncing went from enabled to disabled
-    const isDataSyncStopped =
-      this.#state?.data.isSyncEnabled && !state.data.isSyncEnabled;
+    const dataSyncWasEnabled = this.#state
+      ? this.#state.data.isSyncEnabled
+      : false;
 
-    if (isDataSyncStopped) {
-      this.#maxDataSyncCount = null;
+    // Reset map keeping track of counts used for progress if data sync is toggled
+    if (dataSyncWasEnabled !== state.data.isSyncEnabled) {
+      this.#perDeviceMaxSyncCount.clear();
     } else {
-      const newSyncCount = state.data.want + state.data.wanted;
+      const connectedDevices = Object.keys(state.remoteDeviceSyncState);
+
+      // Remove devices from #perDeviceMaxSyncCount that are no longer found in the new sync state
+      for (const deviceId of this.#perDeviceMaxSyncCount.keys()) {
+        if (!connectedDevices.includes(deviceId)) {
```

Comment:
Nit: this is O(N) but could be O(1).

```suggestion
        if (!Object.hasOwn(state.remoteDeviceSyncState, deviceId)) {
```

(Is this the first time Big O notation has ever been relevant in my career? For this tiny nitpick?)

Reply:
addressed via https://github.com/digidem/comapeo-mobile/pull/619/commits/235dd9bbec5c2160d2d64226981be7a74d7e431a